src/vs/workbench/api/browser/mainThreadInlineChat.ts

```diff
@@ -44,6 +58,7 @@ export class MainThreadInlineChat implements MainThreadInlineChatShape {
 				return {
 					...session,
 					dispose: () => {
+						onDidChangeEnablement.dispose();
```

Comment:
I believe you want to move this into `$unregister...` because this happens at the end of each session (start, end inline chat).